### PR TITLE
refactor: major SDK improvement — correctness, architecture, security, and testing

### PR DESCRIPTION
**Pull request by fbsobreira**
## Summary

Major refactoring of the GoTRON SDK covering correctness fixes, architectural improvements, security hardening, and comprehensive testing. 70 files changed across 16 completed improvement tasks.

### Address package (`pkg/address`)
- `BigToAddress(b)` → `BigToAddress(b) (Address, error)` — returns error on oversize big.Int
- `HexToAddress(s)` → `HexToAddress(s) (Address, error)` — returns error on invalid hex
- `Bytes()` — now returns a copy to prevent external mutation
- `IsValid()` — now validates Base58Check checksum (not just length/prefix)
- `Scan()` — copies incoming buffer instead of aliasing

### Account package (`pkg/account`)
- `ExportPrivateKey`, `ExportKeystore`, `CreateNewLocalAccount`, `ImportFromPrivateKey`, `ImportKeyStore` now accept `Store` as first parameter
- `IsValidPassphrase` enforces minimum 8 characters
- `ExportPrivateKey` returns key as string instead of printing to stdout

### Keys / Mnemonic
- `keys.FromMnemonicSeedAndPassphrase` moved to `mnemonic.FromSeedAndPassphrase` (old kept as deprecated wrapper)

### Ledger (`pkg/ledger`)
- `Device` interface replaces singleton + `log.Fatalln` pattern
- `GetAddress()` / `SignTx()` deprecated — use `OpenDevice()` + `Device` interface
- `ProcessAddressCommand()` removed

### Builders
- `txbuilder.Tx` is single-use — second `Build`/`Send` returns `ErrAlreadyBuilt`
- `contract.SetError` accumulates errors via `errors.Join` (was first-error-wins)

### Dependencies
- `github.com/pkg/errors` removed — all wrapping uses stdlib `fmt.Errorf` with `%w`

## New Packages

| Package | Purpose |
|---------|---------|
| `pkg/txcore` | Shared `Send`/`SendAndConfirm`/`TransactionID`/`Receipt` for both builders |
| `pkg/standards/trc20enc` | Canonical TRC20 ABI encoding (cycle-free, importable from any layer) |
| `pkg/mnemonic` | `FromSeedAndPassphrase` — single mnemonic derivation entry point |

## New Features

- **Service interfaces**: 11 domain-specific interfaces on `GrpcClient` (`Account()`, `Contract()`, `TRC20()`, etc.)
- **Functional options**: `NewGrpcClientWithOptions(addr, WithTimeout, WithAPIKey, WithContext)`
- **CLI: freezeV2 / unfreezeV2**: New commands using `FreezeBalanceV2` API
- **CLI: .env support**: Auto-loads `.env` from current directory via godotenv
- **Environment variables**: `TRONCTL_NODE`, `TRONCTL_KS_DIR`, `TRONCTL_TLS`, `TRONCTL_SIGNER`, `TRONGRID_APIKEY`
- **Store struct**: `store.NewStore(configDir)` with explicit config directory (backward-compat wrappers kept)
- **DebugConfig**: `common.DebugFromEnv()` struct-based debug config
- **keys.ZeroPrivateKey**: Secure key material cleanup
- **Typed sentinel errors**: `ErrInvalidAddress`, `ErrOversizeBigInt`, `ErrNoFromAddress`, `ErrAlreadyBuilt`, etc.
- **CLI Runtime struct**: Replaces scattered globals with focused setup methods

## Security

- Keystore: key zeroization on `Lock`, `TimedUnlock`, `Export`, `Update`
- `writeToFile`: permissions tightened to `0700`/`0600`, removed `os.Chdir`
- Path traversal validation in `ImportFromPrivateKey`
- Ledger: all `panic()` and `log.Fatalln` converted to error returns

## Testing

- **Benchmarks**: address encoding, base58, ABI parsing
- **Fuzz tests**: address, ABI, base58, hexutils
- **Race tests**: keystore concurrent sign/lock, store concurrent access
- **Security tests**: key zeroization, concurrent sign, idempotent close
- **Integration**: `scripts/integration-test.sh` — 36-test round-trip on local TRE node
- **Local node**: `scripts/local-node.sh` — Docker-based TRON node management

## Bug Fixes

- Float precision: `math.Round` for TRX-to-SUN conversion (5 call sites)
- `for range`: iterate values not indices in `generateName` (Go 1.22+)
- Fixed typo: "Blokchain" → "Blockchain"

## Migration Guide

```go
// Address functions now return errors
addr, err := address.BigToAddress(bi)    // was: addr := address.BigToAddress(bi)
addr, err := address.HexToAddress(hex)   // was: addr := address.HexToAddress(hex)

// Account functions accept Store parameter
s := account.DefaultStore()
key, err := account.ExportPrivateKey(s, addr, pass)  // was: account.ExportPrivateKey(addr, pass)

// Ledger uses Device interface
dev, err := ledger.OpenDevice()
defer dev.Close()
addr, err := dev.GetAddress()            // was: addr := ledger.GetAddress()
sig, err := dev.SignTransaction(ctx, tx) // was: sig, err := ledger.SignTx(tx)

// Mnemonic derivation
sk, pk := mnemonic.FromSeedAndPassphrase(m, pass, 0)  // was: keys.FromMnemonicSeedAndPassphrase(...)
```

## Test plan
- [x] `go build ./...` passes
- [x] `go test ./...` — all 21 packages pass
- [x] `make lint` — 0 issues
- [x] `scripts/integration-test.sh` — 36/36 pass on local TRE node
- [x] Multi-agent code review (PR-T + Codex + Security Auditor)

## Summary by CodeRabbit

* **New Features**
  * .env template for node/TLS/API key/signer/keystore/debug flags
  * freezeV2 / unfreezeV2 account commands
  * Improved Ledger device handling and a Ledger mock for testing
  * CLI help text and runtime configuration loading

* **Bug Fixes**
  * Corrected TRX rounding to avoid precision loss
  * Hardened address parsing; defensive copies to prevent surprises
  * Replaced panics with errors for hardware/APDU read edge cases

* **Security Improvements**
  * Path‑validation for imported account names
  * Zeroization of private keys and stricter keystore file permissions

* **Tests / Tools**
  * Extensive fuzz/bench/concurrency/security tests plus integration and local-node scripts

### CODE REVIEW

File: cmd/subcommands/custom-flags.go [hunk 1]
```diff
@@ -16,7 +17,7 @@ func (tronAddress tronAddress) String() string {
 func (tronAddress *tronAddress) Set(s string) error {
 	_, err := address.Base58ToAddress(s)
 	if err != nil {
-		return errors.Wrap(err, "not a valid one address")
+		return fmt.Errorf("not a valid one address: %w", err)
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply cmd/subcommands/custom-flags.go]: Fixed — changed to "not a valid TRON address".

File: pkg/account/export.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ func ExportPrivateKey(address, passphrase string) error {
 			return err
 		}
 		fmt.Printf("%064x\n", key.PrivateKey.D)
+		key.PrivateKey.D.SetInt64(0)
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply pkg/account/export.go]: Good catch — `SetInt64(0)` only changes the logical value, the backing memory persists. Added `keys.ZeroECDSAKey()` which zeros via `D.Bits()` (same pattern as keystore's internal `zeroKey`), and updated `export.go` to use it.
